lib/internal/readline/interface.js

```diff
@@ -449,8 +454,15 @@ class Interface extends InterfaceConstructor {
 
     // If the trimmed line is empty then return the line
     if (StringPrototypeTrim(this.line).length === 0) return this.line;
-
-    if (this.history.length === 0 || this.history[0] !== this.line) {
+    const normalizedLine = StringPrototypeReplaceAll(this.line, '\n', '\r');
+
+    if (this.history.length === 0 || this.history[0] !== normalizedLine) {
+      if (this.lastCommandErrored && this.historyIndex === 0) {
+        // If the last command errored, remove it from history.
+        // The user is issuing a new command starting from the errored command,
+        // Hopefully with the fix
+        ArrayPrototypeShift(this.history);
+      }
```

Comment:
I definitely like the change ❤️

I believe it's worth noting it in the repl docs as changes part and we should probably generally check if the documentation needs an update with these changes.